cmake/nuttx/px4_impl_nuttx.cmake

```diff
@@ -248,43 +248,48 @@ function(px4_nuttx_add_export)
 		REQUIRED OUT CONFIG THREADS
 		ARGN ${ARGN})
 
-	set(nuttx_src ${PX4_BINARY_DIR}/${CONFIG}/NuttX)
+	set(nuttx_build_src ${PX4_BINARY_DIR}/${CONFIG}/NuttX)
+	set(nuttx_export_dir ${nuttx_build_src}/nuttx/nuttx-export)
 
-	# all patches
+	# nuttx cmake dependency files
+	set(nuttx_copy ${PX4_BINARY_DIR}/${CONFIG}_nuttx_copy.stamp)
+	set(nuttx_configure ${PX4_BINARY_DIR}/${CONFIG}_nuttx_configure.stamp)
+	set(nuttx_export ${PX4_BINARY_DIR}/${CONFIG}_nuttx_export.stamp)
+
+	# all nuttx patches
 	file(GLOB nuttx_patches ${PX4_SOURCE_DIR}/nuttx-patches/*.patch)
 	list(SORT nuttx_patches)
 
 	# copy
-	file(GLOB_RECURSE nuttx_all_files ${PX4_SOURCE_DIR}/NuttX/*)
+	file(GLOB_RECURSE nuttx_all_files ${PX4_SOURCE_DIR}/NuttX/nuttx/Makefile)
 	file(RELATIVE_PATH nuttx_cp_src ${PX4_BINARY_DIR} ${PX4_SOURCE_DIR}/NuttX)
-	add_custom_command(OUTPUT ${PX4_BINARY_DIR}/nuttx_copy_${CONFIG}.stamp
-		COMMAND ${MKDIR} -p ${nuttx_src}
-		COMMAND rsync -a --delete --exclude=.git ${nuttx_cp_src}/ ${CONFIG}/NuttX/
-		COMMAND ${TOUCH} ${PX4_BINARY_DIR}/nuttx_copy_${CONFIG}.stamp
-		DEPENDS ${DEPENDS} ${nuttx_patches} ${nuttx_all_files}
+	add_custom_command(OUTPUT ${nuttx_copy}
+		COMMAND ${MKDIR} -p ${nuttx_build_src}
+		COMMAND rsync -rp --delete --exclude=.git --exclude=nuttx-export ${nuttx_cp_src}/ ${CONFIG}/NuttX/
+		COMMAND cmake -E touch ${nuttx_copy}
```

Comment:
Can we now have a rebuild with only an update on the changed file if a NuttX src file is edited?